compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/SourceDirectoryManager.java

```diff
@@ -62,10 +68,15 @@ public static SourceDirectoryManager getInstance(CompilerContext context) {
 
     private SourceDirectoryManager(CompilerContext context) {
         context.put(PROJECT_DIR_KEY, this);
+        // We will add the source directory manager as the project API to the context
+        // This has to be further refactored with ProjectAPI Implementation
+        context.put(PROJECT_KEY, this);
 
         this.names = Names.getInstance(context);
         this.options = CompilerOptions.getInstance(context);
         this.sourceDirectory = initializeAndGetSourceDirectory(context);
+        // load manifest only once
+        this.manifest = getManifest();
```

Comment:
getManifest() logic can be improved to load manifest only once.

Reply:
Created #24516 to track the issue. 

I will be merging this PR without this refactor since that change will touch many places. Will address it with the new Project API